core/src/main/java/org/prevayler/foundation/Chunking.java

```diff
@@ -23,11 +23,11 @@ private static boolean validToken(String token) {
 
   public static void writeChunk(OutputStream stream, Chunk chunk) throws IOException {
     stream.write(Integer.toHexString(chunk.getBytes().length).toUpperCase().getBytes(ASCII));
-    Iterator iterator = chunk.getParameters().entrySet().iterator();
+    Iterator<Map.Entry<String, String>> iterator = chunk.getParameters().entrySet().iterator();
     while (iterator.hasNext()) {
-      Map.Entry entry = (Map.Entry) iterator.next();
-      String name = (String) entry.getKey();
-      String value = (String) entry.getValue();
+      Map.Entry<String, String> entry = iterator.next();
```

Comment:
Nice! I didn't realize we had so many raw types left.

Reply:
More like this will be coming soon; see my comment above

Reply:
Please see the recent pushes on my cloned repo.